### PR TITLE
Alerting: Add toggle to disable alert threshold visibility in graph panel

### LINKED ISSUE #7258: [Feature request] Make alert threshold visibility in graphs optional
Hey 👋 

For my use-cases I don't really need to see the thresholds of an alert in my graphs and they're pretty stark visually (red and the whole alertable area is tinted as well). That's why I was thinking that perhaps it would make sense to make it optional. 

Maybe a checkbox "Show alert thresholds" in "Alert Config"? Or does "Display > Thresholds" make more sense for this?

### PR DESCRIPTION
**Pull request by jpalpant**
Credit to @ikkemaniac in #25034, but taking this to sign CLA and get it landed. Four commit message were:

- add 'fill' switch on Alert Tab
- add 'fill' to Alert init model. Make default value 'true' not to break current design
- use newly created alert.fill when rendering graph
- add 'line' switch on Alert tab, add 'line' to Alert init model. Set default to 'true' not to break current design. Use newly created alert.line when rendering graph

**What this PR does / why we need it**: Alert thresholds and regions can be ugly, filling the whole graph with red and hiding useful information. We should be able to turn them off.

**Which issue(s) this PR fixes**:

Fixes #7258

Samples from trying this out in devenv

Outside-of-range alert showing thresholds:
Screenshot 2020-06-30 00 57 35

Then again with thresholds disabled:
Screenshot 2020-06-30 00 57 43

The edit mode view, with thresholds enabled (the default):
Screen Shot 2020-06-30 at 12 59 20 AM

With the toggle disabled:
Screen Shot 2020-06-30 at 12 59 36 AM

Toggle goes below the Point radius toggle:
Screenshot 2020-06-30 01 02 30

It works as well for other alert types, though I noticed that alerts with compound clauses (BELOW x and ABOVE y) don't have these regions displayed very well - they just show the region of the first clause. That's the case in master though, so not a regression.

**Special notes for your reviewer**:

First time forking grafana; I'm happy to test and make changes as needed, and may need some help with styling tying things together!

### CODE REVIEW

**Comment by torkelo** [comment]
We are planning to move alerts away from the dashboard & graph panel and make them standalone with a separate UI and workflow. 

so having graph only display options in the alert rule is not something that feels right. This could possibly be a graph option to disable thresholds even when an alert rule is present (as a temporary solution until we move alert rules out of graph & dashboard)

**Review by mckn — CHANGES_REQUESTED** [review]
Great work and thanks for helping out fulfilling this feature request! But as @torkelo mentioned we are planning on moving away from the _Alert_ tab so adding this as a display option for the Graph panel might be a better approach.

Skärmavbild 2020-06-25 kl  17 00 26

Please give me a ping if you need any help or have any questions about how to add them as display options.

**Comment by jpalpant** [comment]
Thanks for the feedback! I think I understand and will give it a shot.

**Comment by jpalpant** [comment]
@mckn I believe I've made the change requested, and I've updated the description accordingly. I don't really like bringing `ThresholdMapper` into `GraphCtrl` and sticking all that extra code into every `render()`, but I'm not really sure what my options are here to get things live-updating. What do you think of this?

**Comment by jpalpant** [comment]
Typecheck will also require me to add `alertThreshold` to `PanelModel` as it stands, which I also don't love - would it be better if I make this a property in `PanelModel.options`?

**Comment by dprokop** [comment]
> Typecheck will also require me to add alertThreshold to PanelModel as it stands, which I also don't love - would it be better if I make this a property in PanelModel.options?

I think adding this property to graph panel options (as i.e. dataLinks are specified) is a way to go here. WDYT @mckn?

**Comment by jpalpant** [comment]
Any further feedback on this PR @dprokop @mckn? I see I have some merge conflicts and dataLinks was removed from the options field - I think that is okay and am content with what this looks like now, if you are. And I do have an e2e test issue - I don't really know what to do about that, it doesn't look like the failures are related. Should I rebase or backmerge master, to see if it's fixed?

**Comment by mckn** [comment]
> Any further feedback on this PR @dprokop @mckn? I see I have some merge conflicts and dataLinks was removed from the options field - I think that is okay and am content with what this looks like now, if you are. And I do have an e2e test issue - I don't really know what to do about that, it doesn't look like the failures are related. Should I rebase or backmerge master, to see if it's fixed?

I think it looks really good! Lets see if you can sync your branch with our master to resolve the conflicts. Hopefully that will resolve the e2e tests as well.

**Review by mckn — APPROVED** [review]
LGTM! Sync with master so it can be merged.

File: docs/sources/panels/visualizations/graph-panel.md
Comment (oddlittlebird) [inline docs/sources/panels/visualizations/graph-panel.md:31]:
```suggestion
- **Fill gradient -** Degree of gradient on the area fill. (0 is no gradient, 10 is a steep gradient. Default is XX.
```

Comment (oddlittlebird) [reply docs/sources/panels/visualizations/graph-panel.md:31]: Please enter the default value.

**Review by oddlittlebird — CHANGES_REQUESTED** [review]
A couple minor doc edits.

File: docs/sources/panels/visualizations/graph-panel.md
Comment (jpalpant) [reply docs/sources/panels/visualizations/graph-panel.md:31]: Thanks for the feedback, will incorporate!

**Comment by jpalpant** [comment]
Looks like e2e tests are still failing after I merged master. Going to update with the doc review feedback and take a look later, but if anyone has guidance on if there's an obvious reason for these, I'd appreciate it. Nothing here looks familiar to me 😬

## 1-smoketests.spec.ts 
```
Running:  1-smoketests.spec.ts                                                           (1 of 11)

... logs

  1) Smoke tests
 
       Login scenario, create test data source, dashboard, panel, and export scenario:
 
     AssertionError: Timed out retrying: Expected to find element: `[aria-label="gpl alias A-series"]`, but never found it.
      at scenario (http://localhost:3001/__cypress/tests?p=../../e2e/suite1/specs/1-smoketests.spec.ts:1321:86)
      at Context.eval (http://localhost:3001/__cypress/tests?p=../../e2e/suite1/specs/1-smoketests.spec.ts:1195:45)
```
                                                                                                   
## dashboard-time-zone.spec.ts
```
Running:  dashboard-time-zone.spec.ts                                                    (3 of 11)
 
... Logs showing login succeeding
 
Retrieving Selector: Panel container title Random walk series
 
Retrieving Selector: div.flot-x-axis > div.flot-tick-label
 
    1) Tests dashboard time zone scenarios
 
    ✓ temporary
 
  1 passing (7s)
 
  1 failing
 
  1) Dashboard time zone support
 
       Tests dashboard time zone scenarios:
 
     AssertionError: Timed out retrying: Expected to find element: `div.flot-x-axis > div.flot-tick-label`, but never found it.
 
      at Context.eval (http://localhost:3001/__cypress/tests?p=../../e2e/suite1/specs/dashboard-time-zone.spec.ts:1337:22)
```

**Comment by kminehart** [comment]
I tried restarting the build for you, I thought maybe it was just a flaky test but it failed again. :thinking:

**Comment by jpalpant** [comment]
Wow, totally forgot about this - it's been a hectic few weeks! Going to circle back and give this another go tomorrow.

**Comment by jpalpant** [comment]
Well, it was a real regression - glad the tests caught it! Fixed now. Please re-review when you can!

**Comment by jpalpant** [comment]
@oddlittlebird and @mckn would you mind taking a look and approving this if you have no further concerns? I think this is ready to merge!

**Review by mckn — APPROVED** [review]
LGTM!

**Review by oddlittlebird — APPROVED** [review]
Looks good! Thanks for contributing to Grafana!

**Comment by ikkemaniac** [comment]
> @oddlittlebird and @mckn would you mind taking a look and approving this if you have no further concerns? I think this is ready to merge!

Thx for the effort!

**Comment by ikkemaniac** [comment]
@jpalpant , it's shipped in v7.2.0-beta1. Thx again.

**Comment by leoswing** [comment]
No related options shown in Grafana 9, any one could help to show the settings position in pannel?

**Comment by gatewayshub** [comment]
Do not find this option in Grafana V9.3.1 any hints to get alerting thresholds into panel?

**Comment by Aliraad99** [comment]
for grafana 9, i managed to disable them from the JSON model code
image

do you think it has other effects?

**Pull request merged**
